### PR TITLE
[1.4.x] Fix for avoiding unnecessary constraint import generation

### PR DESCRIPTION
**Pull request by lnash94**
Tested below connectors  with the command `bal openapi -i <yaml> --mode client`: 
This need to generate constraint import since now it is support for the integer and numbers to have zero as value
- [powertoolsdeveloper.collections](https://github.com/ballerina-platform/openapi-connectors/tree/main/openapi/powertoolsdeveloper.collections)
- [powertoolsdeveloper.data](https://github.com/ballerina-platform/openapi-connectors/tree/main/openapi/powertoolsdeveloper.data)
- [powertoolsdeveloper.datetime](https://github.com/ballerina-platform/openapi-connectors/blob/main/openapi/powertoolsdeveloper.datetime/openapi.yml)
- [powertoolsdeveloper.files](https://github.com/ballerina-platform/openapi-connectors/blob/main/openapi/powertoolsdeveloper.files/openapi.yml)
- [powertoolsdeveloper.finance](https://github.com/ballerina-platform/openapi-connectors/blob/main/openapi/powertoolsdeveloper.finance/openapi.yml)
- [powertoolsdeveloper.math](https://github.com/ballerina-platform/openapi-connectors/blob/main/openapi/powertoolsdeveloper.math/openapi.yml)
- [powertoolsdeveloper.text](https://github.com/ballerina-platform/openapi-connectors/blob/main/openapi/powertoolsdeveloper.text/openapi.yml)
- [powertoolsdeveloper.weather](https://github.com/ballerina-platform/openapi-connectors/blob/main/openapi/powertoolsdeveloper.weather/openapi.yml)

These connectors are not included in the constraint import since their swagger includes the wrong constraint https://github.com/ballerina-platform/openapi-tools/issues/1291#issuecomment-1495841381  
- [sinch.verification](https://github.com/ballerina-platform/openapi-connectors/blob/main/openapi/sinch.verification/openapi.yaml)
- [xero.bankfeeds](https://github.com/ballerina-platform/openapi-connectors/blob/main/openapi/xero.bankfeeds/openapi.yml#L627)
- [xero.Project](https://github.com/ballerina-platform/openapi-connectors/blob/main/openapi/xero.projects/openapi.yaml#L1469)

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: openapi-cli/src/test/java/io/ballerina/openapi/generators/schema/ConstraintTests.java [hunk 1]
```diff
@@ -134,4 +134,46 @@ public void testAdditionalPropertiesWithConstraint() throws IOException, Balleri
     }
     //TODO current tool doesn't handle union type: therefore union type constraint will handle once union type
     // generation available in tool.
+
+    @Test(description = "Test for invalid constraint value")
+    public void invalidConstraintUses() throws IOException, BallerinaOpenApiException, FormatterException {
```
Comment (NipunaRanasinghe) [inline openapi-cli/src/test/java/io/ballerina/openapi/generators/schema/ConstraintTests.java:139]: Lets add the `test` prefix for all the new cases to be consistent with the old tests

File: openapi-core/src/main/java/io/ballerina/openapi/core/generators/schema/BallerinaTypesGenerator.java [hunk 2]
```diff
@@ -145,25 +144,25 @@ public SyntaxTree generateSyntaxTree() throws BallerinaOpenApiException {
     }
 
     private NodeList<ImportDeclarationNode> generateImportNodes() {
-        List<ImportDeclarationNode> imports = new ArrayList<>();
+        Set<ImportDeclarationNode> finalImports = new LinkedHashSet<>();
```
Comment (NipunaRanasinghe) [inline openapi-core/src/main/java/io/ballerina/openapi/core/generators/schema/BallerinaTypesGenerator.java:147]:
```suggestion
        Set<ImportDeclarationNode> importDeclarationNodes = new LinkedHashSet<>();
```

Comment (lnash94) [reply openapi-core/src/main/java/io/ballerina/openapi/core/generators/schema/BallerinaTypesGenerator.java]: Addressed with https://github.com/ballerina-platform/openapi-tools/pull/1303/commits/10b5706aa002293a9a4875f908dcb3595a2067fe

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

**Review by aneeshafedo — APPROVED** [review]

**Pull request merged**